components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -347,6 +353,67 @@ fn pow(lhs: &Option<Real>, rhs: &Option<Real>) -> Result<Option<Real>> {
     }
 }
 
+#[inline]
+#[rpn_fn(capture = [metadata],metadata_ctor = init_rng_data)]
+fn rand(metadata: &Option<RefCell<XorShiftRng>>) -> Result<Option<Real>> {
```

Comment:
I think you may not want to merge rand in this PR? Because sre-bot is stupid enough to unable to give you 100 points in single PR.